db/schema/schema.sql

```diff
@@ -0,0 +1,27 @@
+/* ---- TOKENS ---- */
+
+CREATE TABLE token
+(
+    name TEXT NOT NULL UNIQUE
+);
+
+CREATE TABLE token_unit
+(
+    token_name TEXT NOT NULL REFERENCES token (name),
+    denom      TEXT NOT NULL UNIQUE,
+    ibc_denom  TEXT UNIQUE,
+    exponent   INT  NOT NULL,
+    price_id   TEXT
+);
```

Comment:
Based on this table, you are assuming that there can only be one token with a single denom, and that an IBC denom can be present only one. I don't think that these two things separated make sense based on how IBC token denoms work. 

Let's take as an example DSM: if it is moved from the Desmos chain to **two different chains** (eg. Osmosis and Cosmos Hub), then it would have two different denoms (one for each chain). Moreover, if one token is traded on the same chain but using **two different channels**, also two different denoms will be used. 

The denom of an IBC token is based on which channel it's used to transfer it to the recipient chain. For this reason, I think it might be better to redefine this scheme to be the following

```suggestion
CREATE TABLE token_unit
(
    token_name TEXT NOT NULL REFERENCES token (name),
    denom      TEXT NOT NULL,
    ibc_denom  TEXT UNIQUE,
    exponent   INT  NOT NULL,
    price_id   TEXT,
    CONSTRAINT unique_ibc_token_denom UNIQUE (denom, ibc_denom)
);
```

This way we are allowing for multiple `denom - ibc_denom` pairs without any problem and, at the same time, we are making sure that the same `ibc_denom` cannot be used multiple times.

Reply:
Okay but if we remove UNIQUE from denom colum we will loose the column reference inside `token_price` table. What do you think about if we keep `denom` and `ibc_denom` columns UNIQUE, change `ibc_denom` type to []TEXT and store all ibc denoms for given denom inside an array?

Reply:
@MonikaCat I'm not sure how good the support for `[]TEXT` is inside Hasura. I think that implementing it that way might make it impossible to search for IBC denoms using GraphQL. Maybe the best way is to have an additional table: 
```sql
CREATE TABLE token
(
    name TEXT NOT NULL UNIQUE
);

CREATE TABLE token_unit
(
    token_name TEXT NOT NULL REFERENCES token (name),
    denom      TEXT NOT NULL UNIQUE,
    exponent   INT  NOT NULL,
    price_id   TEXT
);

CREATE TABLE token_ibc_denom
(
  token_denom TEXT NOT NULL REFERENCES token_unit (denom),
  ibc_denom   TEXT NOT NULL UNIQUE,
  CONSTRAINT unique_token_ibc_denom UNIQUE (token_denom, ibc_denom)
);
```

This way we can also search for denoms based on a IBC denom easily

Reply:
Cool! I updated the code and hasura metadata as per your suggestion :)